### DISCUSSION TITLE
Suggestion: split favourites in different files

### DISCUSSION THREAD

**Discussion by ThomasMeister123** [osmandapp/OsmAnd#15743] [Ideas]
All favorites are currently stored in ONE XML-file. Even if you create different folders for the favorites and do not display all folders on the map. 

Since I have a favorites folder with >2000 POI, the XML file is very large. This causes OSMAnd (reproducible) to crash. Route is calculating, gets stuck at around 80% and OSMand closes without warning. I think because the huge XML file is parsed. If you had several small files, you would only have to parse the displayed ones

**Comment by vshcherb** [comment]
Yes we plan to split into separate files however Favorites are still not designed to store > 1000 favorites

**Reply by mariush444** [reply]
what is designed to store > 1000 own points?

**Reply by pebogufi** [reply]
Gpx files with wpt.
Use them for points/objects which are not changed frequently.
You can export ("share") a group of favourites just as gpx file with wpt.
For big groups i use it, for me it is more comfortable as using favourites.
Open a gpx file with wpt like a  track.

**Reply by ThomasMeister123** [reply]
Good idea!

**Reply by ThomasMeister123** [reply]
Maybe you can let inspire you by LocusMaps how to implement "large POI-Files" in Osmand 😉 Smaller Icons (just a point), grouping pois,...

**Reply by vshcherb** [reply]
@ThomasMeister123 you already can generate custom Maps using MapCreator and have no limitation on > 1 000 000 points. You can also create huge maps from any GPX and then use as a additional map

**Reply by ThomasMeister123** [reply]
> @ThomasMeister123 you already can generate custom Maps using MapCreator and have no limitation on > 1 000 000 points. You can also create huge maps from any GPX and then use as a additional map

Thanx. Is there a step-by-step-example for GPX somewhere? On https://osmand.net/docs/technical/map-creation/create-offline-maps-yourself/#osmandmapcreator there isnt something about GPX :(

**Reply by mariush444** [reply]
You can use JOSM to convert gpx to pbf then MapCreator can create obf.

**Reply by ThomasMeister123** [reply]
Thanx. But this is like somebody ask "how to drive a car" and the answer is "Buy one, jump in and drive" ;)

What I've done:
- open GPX File in JOSM
- convert to data layer
- saved the new "converted from..." layer as *.osm
- opened MapCreator
- select "create obf from osm"
- copied the obf file on the moble (in the folder with the other obf-mapfiles)
- map is visible in the list of maps

And now? How to show the POIs on the map?

**Reply by ThomasMeister123** [reply]
OK, I got it. I have to give all points a (existing) categorie, an then I have to select these categorie in OSMAnd.
But: Is it possible to create a own categorie? It dont look so.

Now I have a existing (shop/photographie) because the icon fits most for me ;) (its a GPX-file with all places I had visited so far (from GPS-Exif from photos)
